packages/engine/src/simulation/package/state/packages/behavior_execution/package.py

```diff
@@ -26,23 +27,23 @@ def load_behaviors(behavior_descs):
         try:
             # behavior_globals should contain a callable `behavior` if the user's code is correct
             behavior_globals = {}
-            bytecode = compile(desc['source'].decode("utf-8"), desc['name'], "exec")
+            bytecode = compile(desc['source'], desc['name'], "exec")
             exec(bytecode, behavior_globals)
             behavior_fn = behavior_globals.get("behavior")
             
             if callable(behavior_fn):
-                behaviors[hash_behavior_id(desc['id'])] = {
+                behaviors[_hash_behavior_id(desc['id'][0], desc['id'][1])] = {
                     "name": desc['name'],
                     "language": desc['language'],
-                    "required_col_names": desc['columns'],
+                    "required_col_names": None,  # TODO: Propagate desc['columns'] here.
```

Comment:
It's part of loading fewer columns based on behavior keys.

Reply:
Can we change the TODO to better encapsulate that? If someone else tries to pick it up I don't think it's clear.

Also if we don't have an asana for it can you create one and post the link here

Reply:
https://app.asana.com/0/1199548034582004/1201494318833036/f